File: src/mth/nim/DelayedAction.java
package mth.nim;

import javafx.animation.KeyFrame;
import javafx.animation.Timeline;
import javafx.util.Duration;

public class DelayedAction {

    private final Timeline t = new Timeline();

    public DelayedAction(Runnable action, Duration delay) {
        KeyFrame frame = new KeyFrame(delay, e -> action.run());
        t.getKeyFrames().add(frame);
        t.setCycleCount(1);
        t.setAutoReverse(false);
    }

    public void execute() {
        t.playFromStart();
    }

    public static void run(Runnable action, Duration delay) {
        new DelayedAction(action, delay).execute();
    }
}

File: CoreK/src/mth/nim/DelayedAction.kt
package mth.nim

import javafx.animation.KeyFrame
import javafx.animation.Timeline
import javafx.event.ActionEvent
import javafx.util.Duration


class DelayedAction(action: Runnable, delay: Duration) {

    private val t: Timeline

    fun execute() {
        t.playFromStart()
    }

    companion object {
        @JvmStatic
        fun run(action: Runnable, delay: Duration) {
            DelayedAction(action, delay).execute()
        }
    }

    init {
        val frame = KeyFrame(delay, { action.run() })
        t = Timeline().apply {
            keyFrames.add(frame)
            cycleCount = 1
            isAutoReverse = false
        }

    }
}